.github/workflows/build-sample-apps.yml

```diff
@@ -223,11 +235,62 @@ jobs:
           ${{ matrix.sample-app }}-build-main-
 
     - name: Generate SDK binary diff report between main branch and PR
+      id: generate-sdk-size-diff-report
       if: github.event_name == 'pull_request' && matrix.sample-app == 'APN-UIKit'      
       working-directory: Apps/${{ matrix.sample-app }}/build/
       run: |
-        bloaty --source-filter ".*(customerio-ios\/Sources).*" -d compileunits \
-          --debug-file="MainBranchApp.xcarchive/dSYMs/APN UIKit.app.dSYM/Contents/Resources/DWARF/APN UIKit" \
-          --debug-file="App.xcarchive/dSYMs/APN UIKit.app.dSYM/Contents/Resources/DWARF/APN UIKit" \
-          "App.xcarchive/Products/Applications/APN UIKit.app/APN UIKit" -- "MainBranchApp.xcarchive/Products/Applications/APN UIKit.app/APN UIKit"
-  
+        {
+          echo 'report<<EOF'
+          bloaty --source-filter ".*(customerio-ios\/Sources).*" -d compileunits \
+            --debug-file="MainBranchApp.xcarchive/dSYMs/APN UIKit.app.dSYM/Contents/Resources/DWARF/APN UIKit" \
+            --debug-file="App.xcarchive/dSYMs/APN UIKit.app.dSYM/Contents/Resources/DWARF/APN UIKit" \
+            "App.xcarchive/Products/Applications/APN UIKit.app/APN UIKit" -- "MainBranchApp.xcarchive/Products/Applications/APN UIKit.app/APN UIKit"
+          echo EOF
+        } >> "$GITHUB_OUTPUT"
+    
+    - name: Find existing SDK size report comment, if there is one. 
+      if: github.event_name == 'pull_request' && matrix.sample-app == 'APN-UIKit'
+      uses: peter-evans/find-comment@v3
+      id: find-sdk-size-report-comment
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        comment-author: 'github-actions[bot]'
+        body-includes: <!-- sdk size reports -->
+
+    - name: Send SDK size reports to the PR for convenient viewing 
+      if: github.event_name == 'pull_request' && matrix.sample-app == 'APN-UIKit'
```

Comment:
I am wondering that we have hard-coded APN-UIKit as the reference sample app but do we expect the same SDK size reports in FCM sample app ? My question is more around did we do at least one time size check for SDK with FCM implementation? Am I making sense ? 😅

Reply:
Great question. I have generated size reports for the FCM app. Because of the way cocoapods works compared to SPM apps, I found generating the reports to be more work with the FCM app. 

I'll make a comment inside of the Linear ticket talking about this. It would be good for the rest of the team to understand this, too.